### PR TITLE
Change Rules API spec

### PR DESCRIPTION
**Pull request by slavcodev**
- [x] Introduce RuleSetHistoryItem resource
- [x] Change Rule History collection, it contains items resource now
- [x] Add endpoint to get RuleSetVersion entity

### CODE REVIEW

File: spec/definitions/Rules/RuleSetHistoryItem.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  version:
+    type: integer
+  createdTime:
+    $ref: "#/definitions/ServerTimestamp"
+  _links:
+    type: array
+    description: The links related to resource
+    readOnly: true
+    minItems: 1
+    maxItems: 1
```
Comment (XaosSintez) [inline spec/definitions/Rules/RuleSetHistoryItem.yaml]: `maxItems: 2`
And may be minItems is 2 ?

File: spec/paths/custom-events@{id}@rules@history.yaml [hunk 2]
```diff
@@ -4,10 +4,10 @@ get:
   tags:
     - Custom Events
     - Rules
-  summary: Retrieve the history of the set of rules for custom event
+  summary: Retrieve the changes history of the set of rules for custom event
   description: |
-    Retrieve the history versions of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the changes history of the selected set of rules.
+    History are updated each time you change the rules.
```
Comment (XaosSintez) [inline spec/paths/custom-events@{id}@rules@history.yaml]: `History is updated each time you change the rules.`

File: spec/paths/custom-events@{id}@rules@history@{version}.yaml [hunk 3]
```diff
@@ -5,10 +5,10 @@ get:
   tags:
     - Custom Events
     - Rules
-  summary: Retrieve the version of the set of rules for custom event
+  summary: Retrieve the record from changes history of the set of rules for custom event
   description: |
-    Retrieve the version of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the record from changes history of the selected set of rules.
+    History record are created each time you change the rules.
```
Comment (XaosSintez) [inline spec/paths/custom-events@{id}@rules@history@{version}.yaml]: History record is being created each time you change the rules.

File: spec/paths/events@{eventType}@rules@history.yaml [hunk 4]
```diff
@@ -4,10 +4,10 @@ get:
   tags:
     - Events
     - Rules
-  summary: Retrieve the history of the set of rules
+  summary: Retrieve the changes history of the set of rules
   description: |
-    Retrieve the history versions of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the changes history of the selected set of rules.
+    History are updated each time you change the rules.
```
Comment (XaosSintez) [inline spec/paths/events@{eventType}@rules@history.yaml]: Same as above

File: spec/paths/custom-events@{id}@rules@history.yaml [hunk 5]
```diff
@@ -4,10 +4,10 @@ get:
   tags:
     - Custom Events
     - Rules
-  summary: Retrieve the history of the set of rules for custom event
+  summary: Retrieve the changes history of the set of rules for custom event
```
Comment (colinbird) [inline spec/paths/custom-events@{id}@rules@history.yaml]: Retrieve the change history of the set of rules for a custom event

File: spec/definitions/Rules/RuleSetHistoryItem.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+description: Version of rules
+readOnly: true
+properties:
+  version:
+    type: integer
+  createdTime:
+    $ref: "#/definitions/ServerTimestamp"
+  _links:
+    type: array
+    description: The links related to resource
```
Comment (colinbird) [inline spec/definitions/Rules/RuleSetHistoryItem.yaml]: The links related to a resource

File: spec/paths/custom-events@{id}@rules@history@{version}.yaml [hunk 7]
```diff
@@ -5,10 +5,10 @@ get:
   tags:
     - Custom Events
     - Rules
-  summary: Retrieve the version of the set of rules for custom event
+  summary: Retrieve the record from changes history of the set of rules for custom event
```
Comment (colinbird) [inline spec/paths/custom-events@{id}@rules@history@{version}.yaml]: Retrieve the record from the change history of the set of rules for a custom event

File: spec/paths/custom-events@{id}@rules@history@{version}.yaml [hunk 8]
```diff
@@ -5,10 +5,10 @@ get:
...
   description: |
-    Retrieve the version of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the record from changes history of the selected set of rules.
+    History record is created each time you change the rules.
```
Comment (colinbird) [inline spec/paths/custom-events@{id}@rules@history@{version}.yaml]: A history record is created each time you change the rules

File: spec/paths/events@{eventType}@rules@history.yaml [hunk 9]
```diff
@@ -4,10 +4,10 @@ get:
   tags:
     - Events
     - Rules
-  summary: Retrieve the history of the set of rules
+  summary: Retrieve the changes history of the set of rules
   description: |
-    Retrieve the history versions of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the changes history of the selected set of rules.
+    History is updated each time you change the rules.
```
Comment (colinbird) [inline spec/paths/events@{eventType}@rules@history.yaml]: same

File: spec/paths/custom-events@{id}@rules@history@{version}.yaml [hunk 10]
```diff
@@ -5,10 +5,10 @@ get:
   tags:
     - Custom Events
     - Rules
-  summary: Retrieve the version of the set of rules for custom event
+  summary: Retrieve the record from the change history of the set of rules for a custom event
   description: |
-    Retrieve the version of the selected set of rules.
-    Versions are created each time you change the rules.
+    Retrieve the record from the change history of the set of rules for the selected custom event.
+    The history record is created each time you change the rules.
```
Comment (colinbird) [inline spec/paths/custom-events@{id}@rules@history@{version}.yaml]: A history record..?

File: spec/paths/events@{eventType}@rules@history@{version}.yaml [hunk 11]
```diff
@@ -5,16 +5,16 @@ get:
   tags:
     - Events
     - Rules
-  summary: Retrieve the version of the set of rules
+  summary: Retrieve the record from the change history of the set of rules
```
Comment (2e3s) [inline spec/paths/events@{eventType}@rules@history@{version}.yaml]: Do you think this resource `events@{eventType}@rules@history@{version}` is even needed at all? They can query `events@{eventType}@rules@versions@{version}` directly then.

Comment (slavcodev) [inline spec/paths/events@{eventType}@rules@history@{version}.yaml]: Yes, I think it needed. Because of `self` link.

Comment (2e3s) [inline spec/paths/events@{eventType}@rules@history@{version}.yaml]: if for the sake of API consistency, then okay. I was just saying about their real necessity.

**Pull request merged**